Orig address and content type added to headers which are passed to the mapper

BobClaerhout opened the pull request:
Signed-off-by: Bob Claerhout <claerhout.bob@gmail.com>

sophokles73 commented on adapters/mqtt-vertx-base/src/main/java/org/eclipse/hono/adapter/mqtt/impl/HttpBasedMessageMapping.java:
```diff
@@ -121,6 +121,8 @@ private void mapMessageRequest(
             }
         });
+        headers.add(MessageHelper.APP_PROPERTY_ORIG_ADDRESS, ctx.message().topicName());
+        headers.add(MessageHelper.SYS_PROPERTY_CONTENT_TYPE, ctx.contentType());
```
Why don't we use the standard HTTP *Content-Type* header?

BobClaerhout replied:
you're right, the standard HTTP header is better, updated.

sophokles73 commented:
@BobClaerhout not sure if you noticed but the unit tests are failing for your changes ...

BobClaerhout commented:
> @BobClaerhout not sure if you noticed but the unit tests are failing for your changes ...

Fixed it, test now fails due to something with mongo?

sophokles73 commented:
@BobClaerhout there doesn't seem to be any assertions in the unit tests that verify the inclusion of the new properties.
I also think that it would be good to amend the documentation of the mapping feature in the MQTT adapter guide with the properties that are being forwarded to the mapper so that implementers know what to expect. WDYT?

BobClaerhout commented:
> @BobClaerhout there doesn't seem to be any assertions in the unit tests that verify the inclusion of the new properties.
> I also think that it would be good to amend the documentation of the mapping feature in the MQTT adapter guide with the properties that are being forwarded to the mapper so that implementers know what to expect. WDYT?

I updated and added some basic documentation as well. Don't know whether this is enough.

sophokles73 reviewed (approved):
LGTM
